manifests/init.pp

```diff
@@ -1,86 +1,120 @@
+#
+# @param foreman_repo_gpg_chk
+#   Enable / disable GPG checks. Directly passed to Yumrepo resource
+#
+# @param policies
+#   Array of policies that should be configured, each member represent
+#   one policy in form of hash
+#   Policy hash must have following structure
+#
+#   { "id" => 1, "hour" => "*", "minute" => "*", "month" => "*",
+#     "monthday" => "*", "weekday" => "*", "profile_id" => '',
+#     "content_path" => '/usr/share/...',
+#     "tailoring_path" => '/var/lib...'
+#   }
+#
+#   note that profile_id may be empty (for default profile)
+#   id is number representing id of policy in foreman
+#   content_path is path to DS file that contains profile
+#
+#   if it's not array, it's automatically converted to it, so you can
+#   even specify just one policy as a hash
+#
+# @param cron_template
+#   Path to cron template
+#
+# @param cron_splay
+#   Upper limit for splay time when sending reports to proxy
+#
+# @param http_proxy_server
+#   HTTP proxy server
+#
+# @param http_proxy_port
+#   HTTP proxy port
+#
+# @param fetch_remote_resources
+#   Whether client should fetch referenced resources that are remote
+#
+# @param timeout
+#   Timeout when sending reports to proxy
+#
+# @example Run a weekly SCAP audit
+#   class { foreman_scap_client:
+#     server           => 'proxy.example.com',
+#     port             => '8443',
+#     foreman_repo_rel => '1.24',
+#     policies         => [{
+#       "id"                      => 1,
+#       "hour"                    => "12",
+#       "minute"                  => "1",
+#       "month"                   => "*",
+#       "monthday"                => "*",
+#       "weekday"                 => "1",
+#       "profile_id"              => '',
+#       "content_path"            => '/usr/share/xml/scap/ssg/fedora/ssg-fedora-ds.xml',
+#       "download_path"           => '/compliance/policies/1/content',
+#       "tailoring_path"          => '/var/lib/openacap/ssg-fedora-ds-tailored.xml',
+#       "tailoring_download_path" => "/compliance/policies/1/tailoring"
+#     }]
+#   }
 class foreman_scap_client(
-  $server,
-  $port,
-  $policies,
-  $ensure                 = 'present',
-  $fetch_remote_resources = false,
-  $http_proxy_server      = undef,
-  $http_proxy_port        = undef,
-  $ca_file                = $::foreman_scap_client::params::ca_file,
-  $host_certificate       = $::foreman_scap_client::params::host_certificate,
-  $host_private_key       = $::foreman_scap_client::params::host_private_key,
-  $package_name           = $::foreman_scap_client::params::package_name,
-  $foreman_repo_rel       = undef,
-  $foreman_repo_key       = 'https://yum.theforeman.org/RPM-GPG-KEY-foreman',
-  $foreman_repo_src       = undef,
-  $foreman_repo_gpg_chk   = false,
-  $install_options        = undef,
-  $cron_template          = 'foreman_scap_client/cron.erb',
-  $cron_splay             = 600,
-  $timeout                = 60,
+  Stdlib::Host $server,
+  Stdlib::Port $port,
```

Comment:
port used to be a string:

![port-pupp](https://user-images.githubusercontent.com/2664090/70615627-0d35be80-1c0d-11ea-8535-f604632d0f03.png)

We will need [adjustments](https://github.com/theforeman/foreman_openscap/pull/429) on foreman side.